packages/website/src/pipeline/stages/generate-pages/index.ts

```diff
@@ -1,15 +1,191 @@
 import createStage from '../make-pipline-stage';
+import {
+  generatePackageDocPage,
+  generateProjectDocPage,
+  generateDocsHomePage,
+  generateDocumentsMainPage,
+  generateExamplesHomePage,
+  generateExamplePage,
+  generateChangelogPage,
+  generateHomePage,
+  // @ts-ignore: Implicit any
+} from './page-writers';
 
-interface StageInput {}
+interface GenericPage {
+  // The path this page should appear at in the finished website
+  websitePath: string;
+  // Additional data used by the page
+  pageData: object;
+  // Window title for this page
+  title?: string;
+}
 
-// Boilerplate, uncomment when used
-// interface StageConfig {}
+interface DocPage {
+  // The path this page should appear at in the finished website
+  websitePath: string;
+  // An absolute path to the location of the user generated markdown content
+  markdownPath: string;
+  // Additional data used by the page
+  pageData: object;
+  // Metadata about the page
+  meta: object;
+}
 
-export interface StageOutput {}
+interface ExamplePage {
+  // The path the example page should appear at in the finished website
+  websitePath: string;
+  // The path the fullscreen example page should appear at in the finished website
+  fullscreenExampleWebsitePath: string;
+  // An absolute path to the location of the user generated example
+  exampleModulePath: string;
+  // Additional data used by the page
+  pageData: object;
+}
+
+interface ChangelogPage {
+  // The path this page should appear at in the finished website
+  websitePath: string;
+  // An absolute path to the location of the user generated changelogs
+  changelogPath: string;
+  // Additional data used by the page
+  pageData: object;
+  // Window title for this page
+  title?: string;
+}
+
+export interface StageInput {
+  // Absolute path to directory containing page wrapper components
+  wrappersPath: string;
+  // Absolute path to the output pages directory
+  pagesPath: string;
```

Comment:
I'm a fan of dependency injection as a pattern since it makes this code more decoupled from its surroundings (and much easier to test!)

As for wrappers and pages path specifically, I'm expecting that these might fluctuate with our move away from next.js, and having these be passed in lets us flip things as needed.